diff --git a/og.module b/og.module
index 6f28205..0375336 100755
--- a/og.module
+++ b/og.module
@@ -1799,7 +1799,12 @@ function og_membership_label($og_membership) {
  */
 function og_membership_access($op, $entity, $account = NULL, $entity_type = 'og_membership') {
   // No-end user needs access to this entity, so restrict it to admins.
-  return user_access('administer group');
+  if (user_access('administer group')) {
+    return TRUE;
+  }
+
+  // Allow group administrators to modify membership entities.
+  return og_user_access($entity->group_type, $entity->gid, 'manage members');
 }
 
 
